src/zsys.c

```diff
@@ -1071,6 +1078,25 @@ zsys_udp_new (bool routable)
                     (char *) &on, sizeof (on)) == SOCKET_ERROR)
         zsys_socket_error ("setsockopt (SO_REUSEPORT)");
 #endif
+
+    //  Only set TLL for IPv4
+    if (routable) {
```

Comment:
IIRC setting the TTL/HOPS might be blocked and might fail. This should be attempted only if it's > 1 to avoid erroring out for no reason.